st2common/st2common/services/triggers.py

```diff
@@ -46,6 +49,27 @@ def get_trigger_db_given_type_and_params(type=None, parameters=None):
 
         trigger_db = trigger_dbs[0] if len(trigger_dbs) > 0 else None
 
+        # Work around for cron-timer when in some scenarios finding an object fails when Python
+        # value types are unicode :/
+        if not trigger_db and six.PY2 and type == CRON_TIMER_TRIGGER_REF and parameters:
+            non_unicode_literal_parameters = {}
+            for key, value in six.iteritems(parameters):
+                key = key.encode('utf-8')
+                value = value.encode('utf-8')
+                non_unicode_literal_parameters[key] = value
+            parameters = non_unicode_literal_parameters
+
+            trigger_dbs = Trigger.query(type=type,
```

Comment:
If i don't disable query set cache it fails (it will just try to retrieve an item which was previously not found from an internal cache). I basically need to force a cache and only way to do that is by disabling the internal queryset cache.